tf5muxserver/mux_server_PrepareProviderConfig.go

```diff
@@ -42,16 +43,22 @@ func (s muxServer) PrepareProviderConfig(ctx context.Context, req *tfprotov5.Pre
 			resp.Diagnostics = append(resp.Diagnostics, res.Diagnostics...)
 		}
 
-		if res.PreparedConfig != nil {
-			// This could check equality to bypass the error, however
-			// DynamicValue does not implement Equals() and previous mux server
-			// implementations have not requested the enhancement.
-			if resp.PreparedConfig != nil {
-				return nil, fmt.Errorf("got a PrepareProviderConfig PreparedConfig response from multiple servers, not sure which to use")
-			}
+		// Do not check equality on missing PreparedConfig or unset PreparedConfig
+		if res.PreparedConfig == nil {
+			continue
+		}
 
-			resp.PreparedConfig = res.PreparedConfig
+		equal, err := dynamicValueEquals(schemaType(s.providerSchema), res.PreparedConfig, resp.PreparedConfig)
```

Comment:
Correct. My guess is that the previous implementation assumed that terraform-plugin-sdk/v2 (since this came from the older SchemaServerFactory implementation) did not automatically fill in the `PreparedConfig` value, however it does.